dashboard/src/shared/Auth.ts

```diff
@@ -44,6 +44,16 @@ export class Auth {
       }
     }
   }
+
+  public static async fetchToken(): Promise<string | null> {
```

Comment:
I would call this function `fetchOIDCToken` and add a comment explaining that this will make a request to see if an Authorization header is already present